### PR TITLE
Fixed flag exploit by adding safety check

### LINKED ISSUE #409: Fix "Flag capture" exploit
When a person is taking the flag on (for example) a captured outposts he can do it several times whilst effectively in the animation and therefore exploit the system and get sweet money from it.

If I'm correct the function being called is mrkWin.

My idea, change it so the function can only be called once every 2 minutes.
Either you have killed all the enemies in the area and capture the point or you still are outnumbered and have to wait 2 minutes before you can call it again.

This of course should be on server-side and not on client-side to exclude the possibility of others also taking the flag at the same time.

Maybe the system like on the hear-and-repair box.

### PR DESCRIPTION
**Pull request by wurzel0701**
I added a isGettingCaptured variable to the flag, which is getting set to true once someone is capturing the flag. Every further call of the call will abort and hint the player about it.

### CODE REVIEW

File: A3-Antistasi/functions/Base/fn_mrkWIN.sqf [hunk 1]
```diff
@@ -12,32 +12,54 @@ _size = [_markerX] call A3A_fnc_sizeMarker;
...
+		} forEach _revealX;
 		//[_markerX] call A3A_fnc_intelFound;
-		};
 	};
+};
 
 if ((count _revealX) > 2*({([_x,_markerX] call A3A_fnc_canConquer) and (side _x == teamPlayer)} count allUnits)) exitWith {hint "The enemy still outnumber us, check the map and clear the rest of the area"};
 //if (!isServer) exitWith {};
 
+_flagX setVariable ["isGettingCaptured", nil, true];
```
Comment (Spoffy) [inline A3-Antistasi/functions/Base/fn_mrkWIN.sqf]: This still leaves a window for other players to capture it. I'd only set this back to nil *after* we call MarkerChange

File: A3-Antistasi/functions/Base/fn_mrkWIN.sqf [hunk 2]
```diff
@@ -12,32 +12,54 @@ _size = [_markerX] call A3A_fnc_sizeMarker;
...
-		{player reveal _x} forEach _revealX;
+		{
+			player reveal _x
+		} forEach _revealX;
 		//[_markerX] call A3A_fnc_intelFound;
-		};
 	};
+};
 
 if ((count _revealX) > 2*({([_x,_markerX] call A3A_fnc_canConquer) and (side _x == teamPlayer)} count allUnits)) exitWith {hint "The enemy still outnumber us, check the map and clear the rest of the area"};
```
Comment (Spoffy) [inline A3-Antistasi/functions/Base/fn_mrkWIN.sqf]: If we exit here, nobody will ever be able to capture the flag, as we don't nil out the variable during the exit

File: A3-Antistasi/functions/Base/fn_mrkWIN.sqf (hunk 1 above)
Comment (wurzel0701) [reply A3-Antistasi/functions/Base/fn_mrkWIN.sqf]: I added it in markerChange once the new actions are getting added to the flag

File: A3-Antistasi/functions/Base/fn_mrkWIN.sqf (hunk 2 above)
Comment (wurzel0701) [reply A3-Antistasi/functions/Base/fn_mrkWIN.sqf]: Right, damn formating, fixed it

**Comment by wurzel0701** [comment]
Tested and working in SP

**Comment by wurzel0701** [comment]
markerChange can only run once due to 
```sqf
if (_markerX in markersChanging) exitWith {};
markersChanging pushBackUnique _markerX; 
```
The reward for players is given in mrkWin, I added a time limit to it.
The flag can only be taking once in 30 seconds

**Pull request merged**